mods/cef-titlebar-enabler-universal.wh.cpp

```diff
@@ -0,0 +1,259 @@
+cte_offset_t add_child_view_offset = {0, 0, NULL};
+
+typedef int CEF_CALLBACK (*is_frameless_t)(struct _cef_window_delegate_t* self, struct _cef_window_t* window);
+int CEF_CALLBACK is_frameless_hook(struct _cef_window_delegate_t* self, struct _cef_window_t* window) {
+    Wh_Log(L"is_frameless_hook");
+    return 0;
+}
+
+typedef _cef_window_t* (*cef_window_create_top_level_t)(void* delegate);
+cef_window_create_top_level_t cef_window_create_top_level_original;
+_cef_window_t* cef_window_create_top_level_hook(void* delegate) {
+    Wh_Log(L"cef_window_create_top_level_hook");
+
+    if (is_frameless_offset.offset != NULL && cte_settings.showframe == TRUE) {
+        ((is_frameless_t*)((char*)delegate + is_frameless_offset.offset))[0] = is_frameless_hook;
+    }
+    return cef_window_create_top_level_original(delegate);
+}
+
+int cnt = -1;
+
+typedef void CEF_CALLBACK (*add_child_view_t)(struct _cef_panel_t* self, struct _cef_view_t* view);
+add_child_view_t CEF_CALLBACK add_child_view_original;
+void CEF_CALLBACK add_child_view_hook(struct _cef_panel_t* self, struct _cef_view_t* view) {
+    Wh_Log(L"add_child_view_hook: %d", ++cnt);
+    // 0: Minimize, 1: Maximize, 2: Close, 3: Menu (removing this also prevents alt key from working)
+    if (cnt < 3) {
+      if (cte_settings.showcontrols == FALSE) {
+        return;
+      }
+    } else if (cte_settings.showmenu == FALSE) {
+      return;
+    }
+    add_child_view_original(self, view);
+    return;
+}
+
+typedef _cef_panel_t* (*cef_panel_create_t)(void* delegate);
+cef_panel_create_t cef_panel_create_original;
+_cef_panel_t* cef_panel_create_hook(void* delegate) {
+    Wh_Log(L"cef_panel_create_hook");
+    _cef_panel_t* panel = cef_panel_create_original(delegate);
+    if (add_child_view_offset.offset != NULL) {
+        add_child_view_original = ((add_child_view_t*)((char*)panel + add_child_view_offset.offset))[0];
+        ((add_child_view_t*)((char*)panel + add_child_view_offset.offset))[0] = add_child_view_hook;
+    }
+    return panel;
+}
+
+typedef int (*cef_version_info_t)(int entry);
+
+void LoadSettings() {
+    cte_settings.showframe = Wh_GetIntSetting(L"showframe");
+    cte_settings.showmenu = Wh_GetIntSetting(L"showmenu");
+    cte_settings.showcontrols = Wh_GetIntSetting(L"showcontrols");
+}
+
+// The mod is being initialized, load settings, hook functions, and do other
+// initialization stuff if required.
+BOOL Wh_ModInit() {
+    Wh_Log(L"Init");
+    LoadSettings();
+
+    // Check if this process is auxilliary process by checking if the arguments contain --type=
+    int argc;
+    LPWSTR* argv = CommandLineToArgvW(GetCommandLineW(), &argc);
+    BOOL isAuxProcess = FALSE;
+    for (int i = 1; i < argc; i++) {
+        if (wcsstr(argv[i], L"--type=") != NULL) {
+            isAuxProcess = TRUE;
+            break;
+        }
+    }
+    if (isAuxProcess) {
+        Wh_Log(L"Auxilliary process detected, skipping");
+        return TRUE;
```

Comment:
Why not `return FALSE;` here to have the mod unloaded?

Reply:
Changed that to false.

Reply:
@Anixx I think that it's best to return `TRUE` as long as the mod has some effect, and return `FALSE` only if the mod unloads without doing any changes, either due to an error or because the target isn't relevant/supported.